.github/workflows/coverity.yml

```diff
@@ -1,63 +1,81 @@
-#
-# Copyright (C) 2023-2024 Intel Corporation
-#
-# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
-# See LICENSE.TXT
-# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-#
+# Coverity - static analysis build. It requires Coverity's token (set in CI's secret).
 name: coverity-unified-runtime
-# It runs static analysis build - Coverity. It requires special token (set in CI's secret).
 
 on:
   workflow_dispatch:
   schedule:
     # Run every day at 22:00 UTC
     - cron: '0 22 * * *'
 
-env:
-  WORKDIR:                           ${{ github.workspace }}
-  COVERITY_SCAN_NOTIFICATION_EMAIL:  ${{ secrets.COVERITY_SCAN_NOTIFICATION_EMAIL }}
-  COVERITY_SCAN_TOKEN:               ${{ secrets.COVERITY_SCAN_TOKEN }}
-  COVERITY_SCAN_PROJECT_NAME:        ${{ github.repository }}
-  COVERITY_SCAN_BUILD_COMMAND:       "cmake --build ${{github.workspace}}/build"
-  COVERITY_SCAN_BRANCH_PATTERN:      "main"
-  TRAVIS_BRANCH:                     ${{ github.ref_name }}
-
 permissions:
   contents: read
 
 jobs:
-  linux:
+  coverity:
     name: Coverity
-    runs-on: coverity
+    # run only on upstream; forks don't have token for upstream's cov project
+    if: github.repository == 'oneapi-src/unified-memory-framework'
+    runs-on: ubuntu-latest
 
     steps:
-      - name: Clone the git repo
+      - name: Checkout repository
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          fetch-depth: 0
+
+      - name: Install dependencies
+        run: |
+          wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/cuda-keyring_1.1-1_all.deb
+          sudo dpkg -i cuda-keyring_1.1-1_all.deb
+          sudo apt-get update
+          sudo apt-get install -y libhwloc-dev libtbb-dev cuda-toolkit-12-6
 
       - name: Install pip packages
         run: pip install -r third_party/requirements.txt
 
+      - name: Download Coverity
+        run: |
+          wget -O coverity_tool.tgz -nv https://scan.coverity.com/download/linux64 \
+            --post-data "token=${{ secrets.COVERITY_SCAN_TOKEN }}&project=oneapi-src%2Funified-runtime"
+
+      - name: Extract Coverity
+        run: tar xzf coverity_tool.tgz
+
+      # TODO: enable HIP adapter as well (requires proper package(s) installation)
```

Comment:
all packages I tried to install for HIP was too big (storage on public runners should be around ~14GB) - any hints what I could install here to enable HIP adapter's build...?

Reply:
@oneapi-src/unified-runtime-hip-write, any hint what minimal packages I could install that fit within 14GBs and enables UR build...?